### PR TITLE
Mod Categorisation

### LINKED ISSUE #90: Add a Mod Category field (for color coding purposes)
- Create a tag field for a mod created. 
- Based on the prefix of the mod input, automatically add the corresponding category.
- Mainly target the categories required by CS Majors.
- (eg. CS2103T => Computer Science, MA2001 => Maths, ST2334 => Statistics, respective science mods, rest fall under Unrestricted Electives)

### LINKED ISSUE #92: UI for mod category
Create different colors tags for 
- [ ] completed
- [ ] different prefixes
- [ ] interests

To be done after #90 and #86 are merged.

### PR DESCRIPTION
**Pull request by jonasgwt**
**Mod Categories:**
- Computing mods (CS, IS, CP)
- Math mods (ST, MA)
- Science mods (LS, CM, PC)
- Comms mods (ES)
- GE mods (GE, UT)
- UE mods (all others)

Since users are not allowed to modify mod categories, mod categories are not saved in `addressbook.json`, instead, mods are assigned to a category upon its creation. (ie mod categorisation will be handled only within the `Mod` class)

Fixes #90 
Add Mod categorisation
Add tests for mod categorisation
Add more stringent checks for `modName`; mods must now be numbers prefixed with uppercase letters

Minor fixes for #73 
Modify access modifier for `modName` from public to private
Add tests for duplicate mods for `modAddCommand`

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Mod.java [hunk 1]
```diff
@@ -8,11 +8,13 @@
  * Guarantees: immutable; name is valid as declared in {@link #isValidModName(String)}
  */
 public class Mod {
-    public static final String MESSAGE_CONSTRAINTS = "Mod names should not contain any special characters.";
-    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
-
-    public final String modName;
+    public static final String MESSAGE_CONSTRAINTS = "Mod names should be numbers prefixed with letters.";
+    /** Categories for mods */
+    public enum ModCategory { COMP, MATH, SCI, COMMS, GE, UE };
```
Comment (sltsheryl) [inline src/main/java/seedu/address/model/person/Mod.java:13]: I like the use of enum

File: src/test/java/seedu/address/model/person/ModTest.java [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static seedu.address.testutil.Assert.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
 public class ModTest {
+    /**
```
Comment (sltsheryl) [inline src/test/java/seedu/address/model/person/ModTest.java:11]: 👍 Documentation added

**Review by sltsheryl — APPROVED** [review]
Looks good, it passes all the test cases.
Testing is also good. 
I feel we should add the list of mod categories in the user guide.
